### PR TITLE
Enable scala 2.13 (fixing #344)

### LINKED ISSUE #344: scala 2.13
Hi, any chance of a 2.13 build please?

### PR DESCRIPTION
**Pull request by ybasket**
~WIP as it is based on the not yet merged #363 (will rebase asap, for now just ignore the first two commits) and tests didn't compile for native (seems expected), so I let Travis check.~

I found a work-around for https://github.com/scala/bug/issues/11734 (avoiding the indirection via the type aliases in the `squants` package object, see 8cab981). Based on that the PR contains the usual small things to enable the cross build for 2.13, like adapting the build definitions and introducing a small version dependent trait for the tests as `Ordering.FloatOrdering` got split in 2.13.

fixes #344

### CODE REVIEW

**Comment by ybasket** [comment]
@cquiroz could have a look on this one please?

**Comment by Shastick** [comment]
Seconding @ybasket: squants is the _last_ thing holding us back from migrating to 2.13 for several projects :)

File: shared/src/main/scala/squants/motion/Acceleration.scala [hunk 1]
```diff
@@ -8,9 +8,10 @@
 
 package squants.motion
 
-import squants._
-import squants.space.{ Feet, Millimeters, UsMiles }
-import squants.time.{ Seconds, TimeDerivative, TimeIntegral, SecondTimeDerivative, TimeSquared, Time }
+import squants.{AbstractQuantityNumeric, Dimension, PrimaryUnit, Quantity, SiUnit, UnitConverter, UnitOfMeasure}
```
Comment (cquiroz) [inline shared/src/main/scala/squants/motion/Acceleration.scala]: Why these changes? Is it due to the IDE reorganizing it?

File: shared/src/main/scala/squants/energy/package.scala [hunk 2]
```diff
@@ -16,10 +16,10 @@ package squants
 package object energy {
 
   object KineticEnergy {
-    def apply(mass: Mass, velocity: Velocity): Energy =
+    def apply(mass: Mass, velocity: squants.motion.Velocity): Energy =
```
Comment (cquiroz) [inline shared/src/main/scala/squants/energy/package.scala]: Why do we use the full name?

File: shared/src/test/scala/squants/experimental/SquantsNumeric.scala [hunk 3]
```diff
@@ -46,7 +46,7 @@ trait SquantsNumeric[T] extends Ordering[T] {
   implicit def mkSquantsNumericOps(lhs: T): Ops = new Ops(lhs)
 }
 
-object SquantsNumeric {
+object SquantsNumeric extends ScalaVersionSpecificNumericInstances {
```
Comment (cquiroz) [inline shared/src/test/scala/squants/experimental/SquantsNumeric.scala]: Great!

File: shared/src/main/scala/squants/energy/package.scala (hunk 2 above)
Comment (ybasket) [reply shared/src/main/scala/squants/energy/package.scala]: We're in the package `squants`, so the only way to avoid picking up the problematic type aliases in the `squants` companion object are fully qualified names.

File: shared/src/main/scala/squants/motion/Acceleration.scala (hunk 1 above)
Comment (ybasket) [reply shared/src/main/scala/squants/motion/Acceleration.scala]: No, avoiding the import of `squants._` as this was causing the compilation problems due the type aliases defined there. Not beautiful, but has to be like this at least until https://github.com/scala/bug/issues/11734 is resolved.

Comment (cquiroz) [reply shared/src/main/scala/squants/motion/Acceleration.scala]: Ok, thanks for the clarification

**Comment by cquiroz** [comment]
Thanks @ybasket for your PR
I like to wait for approval of another committer before proceeding, perhaps @garyKeorkunian 
Given how important this is, I'll wait just a couple of days before merging and publishing

**Comment by garyKeorkunian** [comment]
LGTM, good work on finding a way around the issue, @ybasket

**Comment by garyKeorkunian** [comment]
If there are no objections, I am OK with merging this.

**Comment by gvolpe** [comment]
Awesome work @ybasket !

@cquiroz what are the plans for a new release? Is there a published snapshot that I can try out?

**Comment by cquiroz** [comment]
I plan to merge some PRs and do a release today

**Comment by gvolpe** [comment]
Great, thanks! :tada:
